plugins/pencilblue/services/authentication/auth_strategy_service.js

```diff
@@ -32,6 +32,13 @@ module.exports = function(pb) {
             delete req.session._loginContext; // Remove login context from session now its used
         };
 
+        // TO DO: this logic is doing something very specific. Please consider refactor for a more general implementation
+        async removeUserExternalUserIdFromSession(session) {
+            if (session && session.authentication && session.authentication.user && session.authentication.user.external_user_id) {
+                delete session.authentication.user.external_user_id;
```

Comment:
I think you can just do: **delete session.authentication.user or session.authentication.user = {} ;** because after the user logs out we don't need any of the data in the "user" property in session.